qiskit_nature/problems/second_quantization/lattice/lattices/lattice.py

```diff
@@ -110,20 +113,27 @@ class Lattice:
     def __init__(self, graph: PyGraph) -> None:
         """
         Args:
-            graph: Input graph for Lattice. `graph.multigraph` must be False.
+            graph: Input graph for Lattice.
 
         Raises:
-            ValueError: If `graph.multigraph` is True for a given graph, it is invalid.
+            ValueError: If the input graph is a multigraph.
+            ValueError: If the graph edges are non-numeric.
         """
         if graph.multigraph:
             raise ValueError(
                 f"Invalid `graph.multigraph` {graph.multigraph} is given. "
                 "`graph.multigraph` must be `False`."
             )
-        if graph.edges() == [None] * graph.num_edges():
-            weighted_edges = [edge + (1.0,) for edge in graph.edge_list()]
-            for start, end, weight in weighted_edges:
-                graph.update_edge(start, end, weight)
+
+        # validate the edge weights
+        for edge_index, edge in graph.edge_index_map().items():
```

Comment:
I'm not familiar with retworkx but I see more changes than just adding the type check here. Can you shortly explain the differences and benefits of your implementation?

Reply:
I just checked out the #472 and I assume that you changed this in order to support both, retworkx and networkx.
If that's the case then it's all good :+1:

Reply:
Yes, partially due to #472 but also to support the case where a single edge weight is `None` but others are not